databroker/assets/handlers.py

```diff
@@ -203,6 +203,105 @@ def __init__(self, filename, frame_per_point=1):
             frame_per_point=frame_per_point)
 
 
+class AreaDetector_HDF5SingleHandler(HandlerBase):
+    '''Handler for hdf5 data stored 1 image per file and returned as a
+    numpy.array.
+
+    Parameters
+    ----------
+    fpath : string
+        filepath
+    template : string
+        filename template string.
+    filename : string
+        filename
+    key : string
+        the 'path' inside the file to the data set.
+    column_names : list[str]
+        The column names of the table
+    frame_per_point : float
+        the number of frames per point.
+    '''
+    specs = {'AD_HDF5_SINGLE'} | HandlerBase.specs
+
+    def __init__(self, fpath, template, filename, key='/entry/data/data',
+                 frame_per_point=1):
+        # I have included defaults for `key` for back compatibility.
+        self._path = os.path.join(fpath, '')
+        self._fpp = frame_per_point
+        self._template = template
+        self._filename = filename
+        self._key = key
+
+    def _fnames_for_point(self, point_number):
+        start = int(point_number * self._fpp)
+        stop = int((point_number + 1) * self._fpp)
```

Comment:
For readability, this can be replaced with `stop = start + self._fpp` (maybe with `int()`).

Reply:
done